### ISSUE TITLE
Admin Customers DOB field

### ISSUE THREAD

**Issue by piloujp** [zencart/zencart#5482]
**Describe the bug**
When updating customer data in Admin a wrong input in dob field will make most of the fields blanks. Those in personal category and phone number stay populated. Plus there is no message indicating input error.

**Version**
Zen Cart version: 1.58 last Github release.
PHP version: 8.1.13 - 8.2.0
MySQL version: 8.0.31
Plugins that may be related: none, fresh install
What older/newer versions were also tested?

**To Reproduce**
Steps to reproduce/demonstrate the behavior:
1. Go to 'admin ->Customers tab -> customers page'
2. Click on 'customer name to edit'
3. input 30 for example in place of month or a letter in place of a number.

**Expected behavior**
A message pointing out the input error but all other field untouched populated with original data.

**Debug/Error Logs**
There two error logs in Zen Cart depending on input error.

number too big:
"#0 [internal function]: zen_debug_error_handler()
#1 D:\wamp\www\testcart\admin\customers.php(725): htmlspecialchars()
#2 D:\wamp\www\testcart\admin\index.php(11): require('...')
--> PHP Deprecated: htmlspecialchars(): Passing null to parameter #1 ($string) of type string is deprecated in D:\wamp\www\testcart\admin\customers.php on line 725."
Repeating for each occurrence of htmlspecialchars(), but this error is not the problem, only consequence of blank fields.

inputing a letter:
"[16-Dec-2022 11:22:28 Asia/Tokyo] PHP Fatal error:  Uncaught TypeError: checkdate(): Argument #2 ($day) must be of type int, string given in D:\wamp\www\testcart\admin\customers.php:145
Stack trace:
#0 D:\wamp\www\testcart\admin\customers.php(145): checkdate()
#1 D:\wamp\www\testcart\admin\index.php(11): require('...')
#2 {main}
  thrown in D:\wamp\www\testcart\admin\customers.php on line 145

[16-Dec-2022 11:22:28 Asia/Tokyo] Request URI: /admin/index.php?cmd=customers&page=1&cID=1&action=update, IP address: -------------------
--> PHP Fatal error: Uncaught TypeError: checkdate(): Argument #2 ($day) must be of type int, string given in D:\wamp\www\testcart\admin\customers.php:145
Stack trace:
#0 D:\wamp\www\testcart\admin\customers.php(145): checkdate()
#1 D:\wamp\www\testcart\admin\index.php(11): require('...')
#2 {main}
  thrown in D:\wamp\www\testcart\admin\customers.php on line 145."
This one can be fix by adding '(int)' in front of substr on lines 143, 144 and 145. Function 'checkdate' needs integer arguments.

**Devices (please complete the following information):**
 - Operating System: Windows 10
 - Browser : Brave v1.46.144 (Dec 14, 2022)

I can't find any indication or error directly related to blank field.

**Comment by mc12345678** [comment]
May I suggest taking a look at for possible application of date checking/validation: https://github.com/zencart/zencart/pull/2337/files

**Comment by piloujp** [comment]
Thanks for the link. Actually I already had a look at it and found date validation interesting. But files structures are so different that I can't make much out of it. customers.php admin file  is more a 'do it all in one' file which makes things more complicated.
I was able to add an error message by replacing line 707 (&nbsp;&nbsp;&nbsp;&nbsp;);?>&nbsp;&nbsp;) by these two lines:
```
        );
	echo($error == true && $entry_date_of_birth_error == true ? '&nbsp;' . ENTRY_DATE_OF_BIRTH_ERROR : '');?>
```
Then adding integer conversion to checkdate arguments everything should work... but no.
Line 394-396 seems to populate fields after error but why half of them I don't know.
```
            } else {
                if ($error == true) {
                    $cInfo = new objectInfo($_POST);
                    $processed = true;
```
Another thing I discovered is that normally populated fields have data from customers table and blank fields should have data from address_book table.... But is it relevant?
Well here is where I am for now.
